src/lib/apimanifest.csproj

```diff
@@ -35,7 +35,7 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="System.Text.Json" Version="[6.0,9.0)" />
+    <PackageReference Include="System.Text.Json" Version="[6.0,10.0)" />
```

Comment:
yes, that's what our team (OpenAPI.net, OpenAPI.net.Odata, this. Microsoft.Plugins.Manifest, kiota, Microsoft.Graph....) has been doing for the past couple of years.
This requires us to be reactive, but gives an escape hatch in case a breaking change would happen with the new version.

Reply:
The doesn't sound ideal if it blocks people from being able to test their apps with .NET 10 _at all_ if you don't relax the restriction until November 2025.

I don't happen to use anything that depends on this assembly, but I test my apps with every preview of .NET before they reach GA. This restriction would be a blocker to any such testing, and it's also not good for users who want to adopt .NET vNext on day 0 and find it to be impossible due to the upper constraint.

Reply:
Does Microsoft.OpenApi have such a restriction? I've had zero issues with it with testing the new Microsoft.AspNetCore.OpenApi package in the .NET 9 previews and the final release. If it did, I feel I'd have opened an issue against it many many months ago about it asking for it to be removed already.

Reply:
Microsoft.OpenAPI is just a set of abstractions, so it does not have any dependencies.
Microsoft.OpenApi.Readers does not depend on STJ (or newtonsoft), but rather [on sharpYAML](https://github.com/microsoft/OpenAPI.NET/blob/9ff9789fff842e9e567dbe70a8a7966014ce8e3d/src/Microsoft.OpenApi.Readers/Microsoft.OpenApi.Readers.csproj#L27C36-L27C45)

This is one of the improvements [of the upcoming v2](https://github.com/microsoft/OpenAPI.NET/blob/863faee69f553623ebbfb0c62ad2dc0dce9fba5f/src/Microsoft.OpenApi.Readers/Microsoft.OpenApi.Readers.csproj#L34) which now makes me realize using a range was forgotten in the process.

Created an issue to address this before the GA https://github.com/microsoft/OpenAPI.NET/issues/1930

Reply:
My issue with this is that it allows _you_ to review any breaking changes, but doesn't allow _anyone else_ to.

This removes a lot of the value to the .NET teams (and yourselves) of customers using your libraries in their own applications and providing feedback over the course of the .NET 10 development lifecycle. If anyone finds an issue you don't find yourself, you won't find out about it until .NET 10 GA, at which point it's "too late".

Reply:
Otherwise, there's a very good chance I'll pop up in February 2025 with an issue saying I can't test any of my OpenAPI-consuming applications with .NET 10 previews and asking for a fix 😃